libbeat/common/match/matcher_bench_test.go

```diff
@@ -0,0 +1,144 @@
+package match
+
+import (
+	"bytes"
+	"fmt"
+	"regexp"
+	"testing"
+)
+
+type testContent struct {
+	name  string
+	lines [][]byte
+}
+
+type benchRunner struct {
+	title string
+	f     func(*testing.B)
+}
+
+var allContents = []testContent{
+	mixedContent,
+	logContent,
+	logContent2,
+	logContentLevel,
+}
+
+var mixedContent = makeContent("mixed", `Lorem ipsum dolor sit amet,
```

Comment:
Very nice to see this extensive test suite. I kind of expect that we will hit some cases we haven't thought of but this will be easily fixible by just adding a test with the exception and find the issue.